scripts/qa_crowbarsetup.sh

```diff
@@ -1648,35 +1648,55 @@ function onadmin_allocate()
         set_node_raid ${controllernodes[0]} $want_raidtype $controller_raid_volumes
     fi
 
-    if [ -n "$want_sles12" ] && iscloudver 5 ; then
+    if [ -n "$want_node_os" ] ; then
+        # OS for nodes provided explicitely: assign them successively to the nodes
+        # example: want_node_os=suse-12.0=3,suse-12.1=4,hyperv-6.3=1
 
-        local nodes=(
-            $(get_all_discovered_nodes | tail -n 2)
-        )
-
-        if [ -n "$deployceph" ] ; then
-            echo "Setting second last node to SLE12 Storage..."
-            set_node_role_and_platform ${nodes[0]} "storage" "suse-12.0"
-        fi
+        local nodesavailable=`get_all_discovered_nodes`
 
-        echo "Setting last node to SLE12 compute..."
-        set_node_role_and_platform ${nodes[1]} "compute" "suse-12.0"
-    fi
+        for systems in ${want_node_os//,/ } ; do
+            node_os=${systems%=*}
+            number=${systems#*=}
 
-    if [ -n "$deployceph" ] && iscloudver 6 ; then
-        local nodes=(
-            $(get_all_discovered_nodes | head -n 3)
-        )
-        for n in $(seq 1 2); do
-            echo "Setting node $(($n+1)) to SLE12 Storage..."
-            set_node_role_and_platform ${nodes[$n]} "storage" "suse-12.0"
+            i=1
+            for node in `printf  "%s\n" $nodesavailable | head -n$number`; do
+                # FIXME or keep this to platform only?
+                local $role = "compute"
```

Comment:
Exactly. Maybe this function should only set the platform, and we'd need another one with the same syntax, assigning roles. Like "want_node_roles=controller=2,storage=3,compute=1".